src/plugins/api.js

```diff
@@ -186,19 +186,43 @@ export async function pushToGitHub(newCommitSha) {
  *
  * An example of the returned JSON structure can be found at 'https://api.github.com/users/adr/repos'
  * @param {number} page
- * @param {number} user - the number of repoistories per page
+ * @param {number} user - the number of repositories per page
  * @returns {Promise<object[]>} the array of repos with attributes 'full_name', 'default_branch', etc.
  */
-export async function loadRepositoryList(page = 1, per_page = 5) {
-    let user = localStorage.getItem("authId");
-    return pizzly
-        .integration("github")
-        .auth(user)
-        .get("/user/repos?sort=updated&page=" + page + "&per_page=" + per_page)
-        .then(response => response.json())
-        .catch(err => {
-            console.log(err);
-        });
+export async function loadRepositoryList(searchText = "", page = 1, per_page = 5) {
+    let auth = localStorage.getItem("authId");
+    console.log("auth: " + auth)
+    if (typeof searchText === "string" && searchText.startsWith("https://github.com/")) {
+        let repoRegExp = new RegExp("https:\\/\\/github\\.com\\/([^/]+)\\/([^/]+)")
+        let match = repoRegExp.exec(searchText)
+        let url = "/repos/" + match[1] + "/" + match[2];
+        console.log("url: " + url);
+        // auth entered URL
+        let repoInfo = pizzly
+            .integration("github")
+            .auth(auth)
+            .get(url)
+            .then(response => {
+                console.log(response)
+                response.json()
+            })
+            .catch(err => {
+                console.log(err);
+                return "[]";
+            });
+        return repoInfo.then(
+            repoInfo => Array.of(repoInfo)
+        )
+    } else {
+        return pizzly
+            .integration("github")
+            .auth(auth)
+            .get("/auth/repos?sort=updated&page=" + page + "&per_page=" + per_page)
```

Comment:
Replace `/auth` by `/user`, i.e., this line should be `.get("/user/repos?sort=updated&page=" + page + "&per_page=" + per_page)`